### PR TITLE
ttl: ttl use timer framework to trigger jobs

### LINKED ISSUE #45468: Migrate TTL job schedule to timer framework

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #45468

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/runtime/runtime.go [hunk 1]
```diff
@@ -115,6 +115,13 @@ func (rt *TimerGroupRuntime) Start() {
 	go rt.loop()
 }
 
+// Running returns whether the runtime is running
+func (rt *TimerGroupRuntime) Running() bool {
```
Comment (bb7133) [inline timer/runtime/runtime.go:119]: Is this function just used for test?

I'm thinking about using RWLock instead, if it is for test only, it is not necessary.

Comment (lcwangchao) [reply timer/runtime/runtime.go:119]: it's only used by test now

**Review by YangKeao — APPROVED** [review]
LGTM

File: timer/runtime/worker.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	workerRecvChanCap               = 8
+	workerRecvChanCap               = 128
```
Comment (bb7133) [inline timer/runtime/worker.go:29]: Any reason to enlarge this?

**Review by bb7133 — COMMENTED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

File: timer/runtime/worker.go (hunk 2 above)
Comment (lcwangchao) [reply timer/runtime/worker.go:29]: To make the timer trigger more smoothness when we have many timers in one hook (.i.e 1000 ttl tables)

**Pull request merged**